### ISSUE TITLE
Test Drive Unlimited Solar Crown Demo (2870020)

### ISSUE THREAD

**Issue by gamingdoom** [ValveSoftware/Proton#7819]
# Compatibility Report
- Name of the game with compatibility issues: Test Drive Unlimited Solar Crown Demo
- Steam AppID of the game: 2870020

## System Information
- GPU: Nvidia RTX 4070 Ti Super
- Video driver version: 555.42.02
- Kernel version: 6.9.3-arch1
- Link to full system information report as [Gist](https://gist.github.com/): [Here](https://gist.github.com/gamingdoom/bf12105f84025806fc6844da5423ee93)
- Proton version: Experimental (experimental-bleeding-edge-9.0-101550-20240606-p124cdc-w2e7379-d8d9653-vb4eb56)

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms 
- The game crashes a few seconds after launch. No windows appear.
- The game crashes with any version of proton.
- No relevant output from the game when steam is run in a terminal emulator.
- The game doesn't crash when using PROTON_LOG, instead it hangs with no window.
[steam-2870020.log](https://github.com/user-attachments/files/15688079/steam-2870020.log)

## Reproduction
1. Launch Game
2. Game Crashes

**Comment by robotman40** [comment]
It seems to be having a problem with ntdll at the end, so is it possibly using a kernel anti-cheat? If so, we're not in good hands.

EDIT: Looking further, it seems to be having issues with the Steam API, so maybe that's another potential issue?

EDIT 2: I'm also seeing errors relating to USB devices and HIDs throughout the logs, so I'm already suspecting that some sort of anti-cheat is being used.

EDIT 3: I just found that AMD AGS is used, so could you try overriding that so the game uses Proton's AMD AGS (add a builtin override for amd_ags_x64.dll)

**Comment by gamingdoom** [comment]
@GalacticalEdge using `WINEDLLOVERRIDES="amd_ags_x64.dll=b" %command%` causes the game to take much longer without any signs of life to crash. During that time it uses CPU and RAM. In addition, it seems to be getting further in the proton logs. The game also doesn't hang anymore when PROTON_LOG is enabled and just crashes.
[steam-2870020.log](https://github.com/user-attachments/files/15689845/steam-2870020.log)

The previous crash seemed to have something to do with defererencing a null pointer since with proton ge, it would output about how the address 0x000...0 was read.

Edit:
The dll override seems to not have anything to do with it. It started taking longer to crash by itself? I can't reproduce the first crash anymore.

**Comment by robotman40** [comment]
> @GalacticalEdge using `WINEDLLOVERRIDES="amd_ags_x64.dll=b" %command%` causes the game to take much longer to crash. In addition, it seems to be getting further in the proton logs. [steam-2870020.log](https://github.com/user-attachments/files/15689845/steam-2870020.log)

Seems to load DXGI now, but seems to encounter issues with it.

**Comment by robotman40** [comment]
Just teated on my Steam Deck and even without adding an AMD AGS Override, the game crashes after a few seconds.

EDIT: Added an override for AMD AGS and no difference.

EDIT 2: Added PROTON_LOG=1 with AMD AGS overridden caused the game to hang and not crash. I did manage to catch some interesting exceptions in the log, such as `RPC_S_SERVER_UNAVAILABLE exception`

**Comment by robotman40** [comment]
Whenever I kill the hanging process, the log ends with `9367.680:012c:0148:err:sync:RtlpWaitForCriticalSection section 00006FFFFFFB0440 "../src-wine/dlls/ntdll/loader.c: peb_lock" wait timed out in thread 0148, blocked by 0130, retrying (60 sec)`

**Comment by gamingdoom** [comment]
> Whenever I kill the process, the log ends with `9367.680:012c:0148:err:sync:RtlpWaitForCriticalSection section 00006FFFFFFB0440 "../src-wine/dlls/ntdll/loader.c: peb_lock" wait timed out in thread 0148, blocked by 0130, retrying (60 sec)`

That was what happened for me in the first crash except it would hang on that and every so often, it would print the same line again.

**Comment by GalacticalEdge** [comment]
@gamingdoom, could that last line where ntdll hangs possibly be related to the crash?

**Comment by gamingdoom** [comment]
> @gamingdoom, could that last line where ntdll hangs possibly be related to the crash?

I don't think so as the hang would only happen with PROTON_LOG. I also can't reproduce that anymore for some reason.

**Comment by robotman40** [comment]
So based off what we know as of now, what seems to be the issue?

Also, to note, the WineUSB errors also appear in other games that exhibit the same symptoms like Call of Duty: Black Ops Cold War, which is why I think the game might have an anti-cheat.

**Comment by gamingdoom** [comment]
> So based off what we know as of now, what seems to be the issue?

I'm not sure. It seems like proton causes some issue with memory. I see multiple SIGSYS in the logs (one at line 1235) and it seems to  dereference a null pointer sometimes

Edit: maybe it actually has something to do with the RPC_S_SERVER_UNAVAILABLE error. [Looking it up online](https://stackoverflow.com/questions/71198239/the-rpc-server-is-unavailable-0x800706ba-win32-1722-rpc-s-server-unavailable), it seems to have something to do with getting a certificate. This then could be failing in a way that would cause bad data/null pointers to eventually be dereferenced.

**Comment by robotman40** [comment]
> > So based off what we know as of now, what seems to be the issue?
> 
> I'm not sure. It seems like proton causes some issue with memory. I see multiple SIGSYS in the logs (one at line 1235) and it seems to dereference a null pointer sometimes
> 
> Edit: maybe it actually has something to do with the RPC_S_SERVER_UNAVAILABLE error. [Looking it up online](https://stackoverflow.com/questions/71198239/the-rpc-server-is-unavailable-0x800706ba-win32-1722-rpc-s-server-unavailable), it seems to have something to do with getting a certificate. This then could be failing in a way that would cause bad data/null pointers to eventually be dereferenced.

If that's the case, maybe a Proton Hotfix could fix it?

**Comment by robotman40** [comment]
Any update on this issue?

**Comment by gamingdoom** [comment]
They updated the game but the same thing is still happening.

**Comment by robotman40** [comment]
> They updated the game but the same thing is still happening.

I mean like any updates to fix this issue on Proton's end?

**Comment by Malthbern** [comment]
> I mean like any updates to fix this issue on Proton's end?

Still no change on both experimental or hotfx

**Comment by robotman40** [comment]
Demo won't be around for much longer and there's still no update

**Comment by robotman40** [comment]
Since the full game is out, I would like to update this saying the game is confirmed to be using an anti-cheat not compatible with Linux at this time. While they said they're working on support, it is not guaranteed to ever come.

But since this game seems to exhibit the same behavior as Call of Duty: Black Ops Cold War, maybe the workaround someone used in its own GitHub issue can be applied to this game?

I don't own the game, so I cannot test this.

**Comment by Malthbern** [comment]
> But since this game seems to exhibit the same behavior as Call of Duty: Black Ops Cold War, maybe the workaround someone used in its own GitHub issue can be applied to this game?

I'm not sure if i performed the workaround you mentioned correctly, however it seems to have not worked. I did get this out of a proton log though

`1090.324:0140:0144:fixme:dbghelp:elf_search_auxv can't find symbol in module`

**Comment by fschaupp** [comment]
> Since the full game is out, I would like to update this saying the game is confirmed to be using an anti-cheat not compatible with Linux at this time. While they said they're working on support, it is not guaranteed to ever come.
> 
> But since this game seems to exhibit the same behavior as Call of Duty: Black Ops Cold War, maybe the workaround someone used in its own GitHub issue can be applied to this game?
> 
> I don't own the game, so I cannot test this.

Where did you get this information from?
Is it public or should we add it to [areweanticheatyet](https://areweanticheatyet.com/game/test-drive-unlimited-solar-crown), so others can also easily join the crowd waiting or at least raising awareness/interest?

**Comment by robotman40** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/7819#issuecomment-2368644642

https://steamdeckhq.com/news/test-drive-unlimited-solar-crown-steam-deck/

**Comment by fschaupp** [comment]
> > Replying to [#7819 (comment)](https://github.com/ValveSoftware/Proton/issues/7819#issuecomment-2368644642)
> 
> https://steamdeckhq.com/news/test-drive-unlimited-solar-crown-steam-deck/

Ok. I hoped, you got it from an (publicly accessible) forum post, social media post or comparable.

**Comment by gofman** [comment]
The just updated Proton Experimental ([bleeding-edge] branch: https://github.com/ValveSoftware/Proton/wiki/Proton-Versions) has some stuff implemented and the game starts with it.

Note that it is completely untested (and I have no idea if anti cheat is fully happy or has a ban potential; I only tried a couple of races a couple of days ago). While the game seems to work in the first glance, might have some issues (will be good to know which ones), one known is that for now it always crashes during exit.

(also I didn't try anything with the demo, only the full game)

**Comment by Malthbern** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/7819#issuecomment-2372716554

Can Confirm that Proton Experimental [bleeding-edge] does indeed work. Time will only tell if this will false positive through anti-cheat or not though.

**Comment by robotman40** [comment]
@kisak-valve, since the discussion has shifted to the full game, should this issue be renamed to reflect this?

**Comment by kisak-valve** [comment]
Hello @robotman40, no, game demos and their full release variants are separate appids and considered separately. If there isn't a compatibility report for the full game release already, then feel free to make one.
